### DISCUSSION TITLE
Can you remove features for a custom, lightweight build?

### DISCUSSION THREAD

**Discussion by Jakobud** [microsoft/monaco-editor#3943] [Q&A]
Monaco seems to be very feature rich. But for my project I need it to be lightweight. Currently it seems like a pretty big payload (several megabytes) especially for delivery over a mobile network.

I don't need most of the features. Is it possible to do a custom build of Monaco where you pick/choose which features are included?

For example I'm wanting to have a lightweight text editor that can have line numbers, word wrapping, scrolling, syntax highlighting and only uses a single language.  Everything else I just don't need. I don't need any of the DIFF functionality, Codicons, text searching functionality, intellisense stuff, keyboard hotkeys, range selection, multi-selection, etc. Just lots of stuff that I don't need that I think seems to be included in the package no matter if I want it or not.

So custom builds... is it possible?

**Comment by hediet** [comment]
Possible, but not easy unfortunately. There is a tree-shaker for the monaco-editor-core (in the VS Code repo) API and a document of all referenced symbols. Maybe you can remove some of them in a fork.
